### ISSUE TITLE
BCPL Events Page Displays Current Day in Two Lists

### ISSUE THREAD

**Issue by jdomasky** [baltimorecounty/BCPL-assets#660]
In its default state, the BCPL Find an Event page displayed the current day and the next day twice, with each day's events split up between two divs. This occurred in the afternoon on April 30 in prod and dev.

**To Reproduce**
1. Go to https://www.bcpl.info/events-and-programs/list.html#!/
2. Do not use the datepicker. If you do, the bug won't occur.
3. Click on 'Load Next 10 Events'
4. Scroll down and observe the date bars
5. Click on 'Load Next 10 Events' again, maybe twice
6. Scroll down to the next day
7. Observe the sequence of the date bars so far: today, tomorrow, today (again), tomorrow (again)

**Expected behavior**
Each day's events should be displayed under a single date bar in a single list, in chronological order.

**Additional context**
The events themselves aren't repeated. The problem is the splitting of the day's events into two separated lists of events.
I didn't see any errors in the browser console.

**Comment by jdomasky** [comment]
FYI, I observed this issue in production today. Although it's been a month since I opened this issue, it remains an active problem on the BCPL website.
On the Events page ( https://www.bcpl.info/events-and-programs/list.html#!/ ), some of the May 28 events are listed, followed by the May 29 events, which are then followed by some May 28 events.

**Comment by danfox01** [comment]
@jdomasky are you only seeing this behavior at the end of a month?

**Comment by jdomasky** [comment]
@danfox01 No. I believe I observed this problem about three times in the past four weeks, but I didn't documented the dates. I'm making a note to confirm the issue on June 7. I'll report my findings.

**Comment by danfox01** [comment]
OK, I thought this related only to dates at the end of the month, which would make it a little less pressing. Since it appears to happen regularly we need to try to hotfix it.

@tmccoy529 can you please take a look at this issue?

**Comment by jdomasky** [comment]
@tmccoy529 @danfox01 This issue doesn't occur all the time. I noticed the issue today around 11:30 a.m. Here is a screen capture:
 
![BCPL Find an Event List with June 7 before and After June 8 - 20190607](https://user-images.githubusercontent.com/13354985/59116679-1a9a8500-891a-11e9-82c7-1d9bfc585faa.png)

**Comment by mcorasan** [comment]
@jdomasky - can you replicate the issue again or have you noticed this recently? I know it was spotty at the time but I'm unable to replicate the issue.

**Comment by jdomasky** [comment]
@mcorasan - I think we need to wait until BCPL starts listing more than 10 events per day. I believe this "splitting" problem occurred after using "Load More" when a single day had more than 10 events. Thankfully, we can be fairly confident that this issue hasn't occurred since the pandemic began.
